src/Stratis.Bitcoin.Features.Wallet/WalletRPCController.cs

```diff
@@ -176,52 +180,140 @@ public decimal GetBalance(string accountName, int minConfirmations = 0)
         }
 
         /// <summary>
-        /// RPC method to return transaction info from the wallet.
-        /// Uses the first wallet and account.
+        /// RPC method to return transaction info from the wallet. Will only work fully if 'txindex' is set.
+        /// Uses the default wallet if specified, or the first wallet found.
         /// </summary>
-        /// <param name="txid">Transaction identifier to find.</param>
+        /// <param name="txid">Identifier of the transaction to find.</param>
         /// <returns>Transaction information.</returns>
         [ActionName("gettransaction")]
-        [ActionDescription("Gets a transaction from the wallet.")]
-        public GetTransactionModel GetTransaction(string txid)
+        [ActionDescription("Get detailed information about an in-wallet transaction.")]
+        public async Task<GetTransactionModel> GetTransactionAsync(string txid)
```

Comment:
So for me personally I would split this method up a bit as it is very long and hard to follow. I just prefer smaller methods and it is easier compartmentalize the logic. It might personal preference so its not a blocker but what do you think?

Reply:
The Fee is something to which we don't have access just yet, but I'm keen to keep it there so that we can review when it's available.

The method is long but it does one thing. Splitting it in many won't help much I'm afraid. I think it even might make it harder to follow.

Reply:
OK cool, if this is the case maybe just add `// TODO: Fee needs to be implemented....`

Reply:
I have it on here
![image](https://user-images.githubusercontent.com/1867877/52635096-dbef4380-2ec0-11e9-8eff-9b45b5f6bfc2.png)

Reply:
I'll add a more meaningful meesage.

Reply:
OK cool, then just add TODO to the tests, so that we know that they also need to be implemented.